core/state-machine/src/backend.rs

```diff
@@ -24,8 +24,17 @@ use log::warn;
 use hash_db::Hasher;
 use crate::trie_backend::TrieBackend;
 use crate::trie_backend_essence::TrieBackendStorage;
-use trie::{TrieDBMut, TrieMut, MemoryDB, trie_root, child_trie_root, default_child_trie_root};
+use trie::{TrieDBMut, TrieMut, MemoryDB, trie_root, child_trie_root, default_child_trie_root, KeySpacedDBMut};
 use heapsize::HeapSizeOf;
+use primitives::subtrie::{KeySpace, SubTrie};
+
```

Comment:
Backend struct renamed as MapTransaction and VecTransaction are really suitable with `SubTrie` struct: especially the usage of `Option<SubTrie>` seems awkward at some points).